// @prisma/fetch-engine/dist/download.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var download_exports = {};
__export(download_exports, {
  download: () => import_chunk_FFV3SNJF.download,
  getBinaryName: () => import_chunk_FFV3SNJF.getBinaryName,
  getVersion: () => import_chunk_FFV3SNJF.getVersion,
  maybeCopyToTmp: () => import_chunk_FFV3SNJF.maybeCopyToTmp,
  plusX: () => import_chunk_FFV3SNJF.plusX,
  vercelPkgPathRegex: () => import_chunk_FFV3SNJF.vercelPkgPathRegex
});
module.exports = __toCommonJS(download_exports);
var import_chunk_FFV3SNJF = require("./chunk-FFV3SNJF.js");
var import_chunk_5564WBCW = require("./chunk-5564WBCW.js");
var import_chunk_7JLQJWOR = require("./chunk-7JLQJWOR.js");
var import_chunk_IF4T5P2G = require("./chunk-IF4T5P2G.js");
var import_chunk_PPPJF3PE = require("./chunk-PPPJF3PE.js");
var import_chunk_VLGS4WJI = require("./chunk-VLGS4WJI.js");
var import_chunk_7AOUSONO = require("./chunk-7AOUSONO.js");
var import_chunk_D2GOUA3T = require("./chunk-D2GOUA3T.js");
var import_chunk_XNR2KDTK = require("./chunk-XNR2KDTK.js");
var import_chunk_66KG5WKV = require("./chunk-66KG5WKV.js");
var import_chunk_FKGWOTGU = require("./chunk-FKGWOTGU.js");
var import_chunk_QTVXEC6H = require("./chunk-QTVXEC6H.js");
var import_chunk_2ESYSVXG = require("./chunk-2ESYSVXG.js");
